conbench/api/history.py

```diff
@@ -42,10 +53,146 @@ def get(self, benchmark_result_id):
         return json_response_for_byte_sequence(jsonbytes, 200)
 
 
+class HistoryDownloadAPI(ApiEndpoint):
+    @maybe_login_required
+    def get(self, benchmark_result_id):
+        """
+        ---
+        description: Download time series
+        responses:
+            "200": "HistoryList"
+            "401": "401"
+            "404": "404"
+        parameters:
+          - name: benchmark_result_id
+            in: path
+            schema:
+                type: string
+        tags:
+          - History
+        """
+        # TODO: think about the case where samples if of zero length. Can this
+        # happen? If it can happen: which response would we want to emit to the
+        # HTTP client? An empty array, or something more convenient?
+        try:
+            items = get_history_for_benchmark(benchmark_result_id=benchmark_result_id)
+        except NotFound:
+            self.abort_404_not_found()
+
+        csv_buf = generate_csv_history_for_result(benchmark_result_id, items)
+
+        return send_file(
+            csv_buf,
+            as_attachment=True,
+            # Use a "timeseries" fingerprint here that represents what all data
+            # points in this series have in common: benchmark name, case perm,
+            # hardware, context, repo.
+            download_name=f"conbench-history-{items[0].benchmark_name}-{items[0].ts_fingerprint}.csv",
+            mimetype="text/csv",
+        )
+
+        # # Note: wrap this into an array if there is just 1 sample, for
+        # # consistency (clients expect an array).
+        # jsonbytes: bytes = orjson.dumps(
+        #     [s._dict_for_api_json() for s in samples],
+        #     option=orjson.OPT_INDENT_2,
+        # )
+        # return csv_response_for_byte_sequence(csv_buf.getbuffer(), 200)
+
+
 history_entity_view = HistoryEntityAPI.as_view("history")
+history_download_endpoint = HistoryDownloadAPI.as_view("history-download")
+
+rule(
+    "/history/download/<benchmark_result_id>/",
+    view_func=history_download_endpoint,
+    methods=["GET"],
+)
```

Comment:
Would it be useful to structure this as a query param, e.g. `http://...?format=csv`?

If csv and json contain different data, the answer may be "no", but that in turn raises the question of why they're different.

Reply:
> Would it be useful to structure this as a query param, e.g. http://...?format=csv?

Maybe! I didn't want to overthink this yet, but I also thought about `/benchmark-results/<id>/histfile?format=<fmt>` where ideally we have a more structured container like parquet or HDF5.

> why they're different.

I think one good (enough) reason is that we can go crazy here and build something for us and not worry about breaking an interface for others.

We can then later zoom out and look at both and think about if and how we want to reconcile this with the current/existing history endpoint.